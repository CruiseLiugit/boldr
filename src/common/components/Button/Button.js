/* @flow */
import React from 'react';

type Props = {
  onClick: () => void,
  children: ReactElement,
  submit: boolean,
  type: String,
  disabled: boolean,
  big: boolean,
  loading: boolean
};

const Button = (props: Props) => {
  const { disabled, loading, big } = props;
  const type = props.submit ? 'submit' : 'button';
  let className = `boldr-btn boldr-btn__${props.type}`;

  if (loading) {
    className += ' boldr-btn__loading';
  }
  if (big) {
    className += ' boldr-btn__big';
  }

  // Unless the disabled state is explicitly set, the button is disabled when loading.
  const isDisabled = (disabled !== undefined ? disabled : loading) || false;

  return (
    <button
      type={ type }
      className={ className }
      disabled={ isDisabled }
      onClick={ props.onClick }
    >
      { props.children }
    </button>
  );
};

Button.defaultProps = {
  type: 'default',
  submit: false,
};

export default Button;
