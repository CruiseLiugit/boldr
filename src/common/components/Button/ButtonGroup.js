/* @flow */
import React from 'react';

type Props = {
  children: ReactElement
};

const ButtonGroup = (props: Props) => {
  return (
    <div className="boldr-btn__group">
      { props.children }
    </div>
  );
};

export default ButtonGroup;
