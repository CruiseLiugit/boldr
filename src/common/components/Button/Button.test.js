import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Button from './Button';

it('<Button />, should render', () => {
  const wrapper = shallow(<Button />);
  const wrapperInner = wrapper.find('.boldr-btn');
  expect(wrapperInner.is('.boldr-btn')).toBe(true);
});

it('renders snapshot', () => {
  const wrapper = shallow(<Button />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
