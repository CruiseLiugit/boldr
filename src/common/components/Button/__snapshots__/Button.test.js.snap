exports[`test renders snapshot 1`] = `
<button
  className="boldr-btn boldr-btn__default"
  disabled={false}
  type="button" />
`;
